api/envoy/api/v2/route/route.proto

```diff
@@ -152,14 +152,22 @@ message VirtualHost {
   google.protobuf.UInt32Value per_request_buffer_limit_bytes = 18;
 }
 
+// A filter-defined action type.
+message FilterAction {
+  oneof action {
+    google.protobuf.Any filter_action = 1;
+    google.protobuf.Struct typed_filter_action = 2;
```

Comment:
Yep, this is what I was thinking. The nifty thing is that it's possible to stuff a serialized `RouteAction` in here, with missing cluster for your use case, so no need to redefine any protos for that.